src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs

```diff
@@ -254,6 +254,12 @@ public async Task<AuthenticationResult> GetAuthenticationResultForUserAsync(
 
             var application = await GetOrBuildConfidentialClientApplicationAsync(mergedOptions);
 
+            if (tokenAcquisitionOptions is not null)
+            {
+                tokenAcquisitionOptions.ExtraParameters ??= new Dictionary<string, object>();
+                tokenAcquisitionOptions.ExtraParameters[Constants.ExtensionOptionsServiceProviderKey] = _serviceProvider;
+            }
+
             try
             {
                 AuthenticationResult? authenticationResult;
```

Comment:
Can you please expose `bool FromCache` property to be able to assert in tests if the token comes from cache or not?